phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/WriteWorkload.java

```diff
@@ -0,0 +1,403 @@
+    private static final Logger logger = LoggerFactory.getLogger(WriteWorkload.class);
+    private final PhoenixUtil pUtil;
+    private final XMLConfigParser parser;
+    private final RulesApplier rulesApplier;
+    private final ResultUtil resultUtil;
+    private final ExecutorService pool;
+    private final WriteParams writeParams;
+    private final Scenario scenario;
+    private final long threadSleepDuration;
+
+    private final int threadPoolSize;
+    private final int batchSize;
+
+    public WriteWorkload(XMLConfigParser parser) throws Exception {
+        this(PhoenixUtil.create(), parser);
+    }
+
+    public WriteWorkload(PhoenixUtil util, XMLConfigParser parser) throws Exception {
+        this(util, parser, null);
+    }
+
+    public WriteWorkload(PhoenixUtil phoenixUtil, XMLConfigParser parser, Scenario scenario)
+            throws Exception {
+        this(phoenixUtil, PherfConstants.create().getProperties(PherfConstants.PHERF_PROPERTIES),
+                parser, scenario);
+    }
+
+    /**
+     * Default the writers to use up all available cores for threads. If writeParams are used in
+     * the config files, they will override the defaults. writeParams are used for read/write mixed
+     * workloads.
+     * TODO extract notion of the scenario list and have 1 write workload per scenario
+     *
+     * @param phoenixUtil {@link org.apache.phoenix.pherf.util.PhoenixUtil} Query helper
+     * @param properties  {@link java.util.Properties} default properties to use
+     * @param parser      {@link org.apache.phoenix.pherf.configuration.XMLConfigParser}
+     * @param scenario    {@link org.apache.phoenix.pherf.configuration.Scenario} If null is passed
+     *                    it will run against all scenarios in the parsers list.
+     * @throws Exception
+     */
+    public WriteWorkload(PhoenixUtil phoenixUtil, Properties properties, XMLConfigParser parser,
+            Scenario scenario) throws Exception {
+        this.pUtil = phoenixUtil;
+        this.parser = parser;
+        this.rulesApplier = new RulesApplier(parser);
+        this.resultUtil = new ResultUtil();
+
+        // Overwrite defaults properties with those given in the configuration. This indicates the
+        // scenario is a R/W mixed workload.
+        if (scenario != null) {
+            this.scenario = scenario;
+            writeParams = scenario.getWriteParams();
+            threadSleepDuration = writeParams.getThreadSleepDuration();
+        } else {
+            writeParams = null;
+            this.scenario = null;
+            threadSleepDuration = 0;
+        }
+
+        int size = Integer.parseInt(properties.getProperty("pherf.default.dataloader.threadpool"));
+
+        this.threadPoolSize = (size == 0) ? Runtime.getRuntime().availableProcessors() : size;
+
+        // TODO Move pool management up to WorkloadExecutor
+        this.pool = Executors.newFixedThreadPool(this.threadPoolSize);
```

Comment:
Is the intent here to have a separate pool for write work load? I am not sure if the MonitorManager is running on this pool too. Would be good to monitor write activity too.